### ISSUE TITLE
Integer overflow in gossip traffic metrics

### ISSUE THREAD

**Issue by lzpap** [iotaledger/goshimmer#686]
### Description
The way we currently measure gossip traffic leads to integer overflow. 

From `plugins/metrics/plugin.go/`:
```go
				// gossip network traffic
				g := gossipCurrentTraffic()
				gossipCurrentRx.Store(uint64(g.BytesRead))
				gossipCurrentTx.Store(uint64(g.BytesWritten))
			}, 1*time.Second, shutdownSignal)
```
`g.BytesRead` and `g.BytesWritten` are both `uint32`, so they overflow at around 4.29 Gigabyte.

### Proposed Solution(s)
1. Change the way we measure gossip traffic in metrics plugin, take into account the expected overflow.

Or:

2. Change `buffconn` package in `hive.go` to use `uint64` for storing the read and written bytes.

I think Option 1 is sufficient, as this is quite low priority anyway.

**Comment by Wollac** [comment]
If we really think that 4GB is not enough, I think we should be consistent and change it in the `buffconn` especially since the corresponding `bytesRead` /  `bytesWritten` in `network.ManagedConnection` is already `uint64`.
Also since we are moving away more and more from 32bit architectures it should not even have any relevant drawbacks.

**Comment by capossele** [comment]
Fixed by #695
